src/qt/walletmodel.h

```diff
@@ -179,7 +179,9 @@ class WalletModel : public QObject
     // Cache some values to be able to detect changes
     interfaces::WalletBalances m_cached_balances;
     EncryptionStatus cachedEncryptionStatus;
-    int cachedNumBlocks;
+
+    // Block hash denoting when the last balance, polling, update was done.
+    mutable std::atomic<uint256> cachedLastBalanceUpdate;
```

Comment:
In commit "[WalletModel] Balance and Txrecord status change polling check based on the block hash and not the height." (997129bdb45c01390bffce991b680d58c9e7d18b)

Again `std::atomic<uint256>` doesn't seem to be a valid type. You probably need something like

```c++
Mutex m_cached_tip_mutex;
uint256 m_cached_tip;
```

Reply:
I'm thinking on this one. 

No synchronization is needed here. Only `pollBalanceChanged` modifies the cached tip, which is launched by the timer only. And QTimer says: If the execution takes longer than the timer schedule, the new event will be queued. So, no parallel code execution.